### ISSUE TITLE
plot_components_plotly malfunctions when using m.add_country_holidays()

### ISSUE THREAD

**Issue by rteuwens** [facebook/prophet#1525]
**method:** `fbprophet.plot.plot_components_plotly()`
**issue:** not compatible with `m.add_country_holidays()`

`holidays` variable does not seem to be a Dataframe, causing an `AttributeError`.

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-50-fce739a52069> in <module>
----> 1 fig = fbprophet.plot.plot_components_plotly(m, forecast)
      2 fig.update_layout(layout)
      3 fig.show()

~\AppData\Local\Continuum\anaconda3\envs\forecast\lib\site-packages\fbprophet\plot.py in plot_components_plotly(m, fcst, uncertainty, plot_cap, figsize)
    713     if m.train_holiday_names is not None and 'holidays' in fcst:
    714         components['holidays'] = get_forecast_component_plotly_props(
--> 715             m, fcst, 'holidays', uncertainty)
    716 
    717     regressors = {'additive': False, 'multiplicative': False}

~\AppData\Local\Continuum\anaconda3\envs\forecast\lib\site-packages\fbprophet\plot.py in get_forecast_component_plotly_props(m, fcst, name, uncertainty, plot_cap)
    837         mode = 'markers'
    838         # Combine holidays into one hover text
--> 839         holiday_features, _, _ = m.make_holiday_features(fcst['ds'], m.holidays)
    840         holiday_features.columns = holiday_features.columns.str.replace('_delim_', '', regex=False)
    841         holiday_features.columns = holiday_features.columns.str.replace('+0', '', regex=False)

~\AppData\Local\Continuum\anaconda3\envs\forecast\lib\site-packages\fbprophet\forecaster.py in make_holiday_features(self, dates, holidays)
    532         row_index = pd.DatetimeIndex(dates.apply(lambda x: x.date()))
    533 
--> 534         for _ix, row in holidays.iterrows():
    535             dt = row.ds.date()
    536             try:

AttributeError: 'NoneType' object has no attribute 'iterrows'
```

**Comment by bletham** [comment]
Thanks for raising this bug.

What's missing is that instead of using `m.holidays` it should first do
```
holidays = m.construct_holiday_dataframe(fcst['ds'])
holiday_features, _, _ = m.make_holiday_features(fcst['ds'], holidays)
```
(which is along the lines of 
https://github.com/facebook/prophet/blob/f16d9df33337be93d34f0f69904e9fa25e1b6a10/python/fbprophet/forecaster.py#L779
)
I think that should be all that is needed to fix this.

**Comment by rteuwens** [comment]
Thanks, @bletham, I did just that and it seems to work.

I would like to change the way the holidays are graphed with plotly, though. The current scatter points have less of an impact than the original component plotting.

![grafik](https://user-images.githubusercontent.com/30375696/83727723-d1bcff80-a645-11ea-87ea-d19d61db98fd.png)

vs

![grafik](https://user-images.githubusercontent.com/30375696/83727810-f5804580-a645-11ea-89d9-e948cb445491.png)


It'll pollute the code a bit, but maybe we can figure out a way to integrate something along the lines of
```python
holidays.set_index('ds', inplace=True)
full_date_range = pd.date_range(min(holidays.index), max(holidays.index))
holidays = holidays.reindex(full_date_range, fill_value=0)
```

**Comment by rteuwens** [comment]
Turned out to be much simpler, by removing:

https://github.com/facebook/prophet/blob/f16d9df33337be93d34f0f69904e9fa25e1b6a10/python/fbprophet/plot.py#L836-L837

![image](https://user-images.githubusercontent.com/30375696/83867222-60567d00-a729-11ea-932d-d022e14e8dd8.png)

Perhaps there is a way to disable the hoverinfo when `y==0`, but I already prefer this over the scatter points.

Pushed to a pull request for you to do with as you please.

**Comment by bletham** [comment]
This fix has been pushed in v0.7.
